Homework-7/script.js

```diff
@@ -0,0 +1,292 @@
+// console.log(age);
+// console.log(isAdmin);
+
+// Task 2
+
+// const salaries = {
+//   John: 100,
+//   Pete: 300,
+//   Mary: 250
+// };
+
+// const topSalary = obj => {
+//   const salary = Object.entries(obj);
+//   if (Array.isArray(salary) && salary.length) {
+//     return salary.reduce((prev, cur) => {
+//       return prev[1] >= cur[1] ? prev : cur;
+//     })[0];
+//   }
+//   return null;
+// }; // <= in my opinion better option
+
+// const topSalary = obj => {
+//   let maxSalary = 0;
+//   let maxName = null;
+//   for (const [name, salary] of Object.entries(obj)) {
+//     if (maxSalary < salary) {
+//       maxSalary = salary;
+//       maxName = name;
+//     }
+//   }
+//   return maxName;
+// };
+
+// console.log(topSalary(salaries));
+
+// Task 3
+
+// alert(new Date('2012-02-20T03:12:00.000+03:00'));
+
+// Task 4
+
+// const date = new Date();
+// const getWeekDay = date => {
+//   const days = ["ВС", "ПН", "ВТ", "СР", "ЧТ", "ПТ", "СБ"];
+//   return days[date.getDay()];
+// };
+// console.log(getWeekDay(date));
+
+// Task 5
+
+// const date = new Date();
+// const getWeekDay = date => {
+//     const days = [7, 1, 2, 3, 4, 5, 6];
+//     return days[date.getDay()];
+
+//   // return date.getDay() === 0 ? 7 : date.getDay();
+// };
+// console.log(getWeekDay(date));
+
+// Task 6
+
+// const date = new Date(2015, 0, 2);
+
+// const getDateAgo = (date, daysAgo) => {
+//   return new Date(date.getTime() - daysAgo * 24 * 60 * 60 * 1000).getDate();
+// };
+
+// console.log(getDateAgo(date, 1));
+// console.log(getDateAgo(date, 2));
+// console.log(getDateAgo(date, 365));
+
+// Task 7
+
+// const getLastDayOfMonth = (year, month) => {
+// //   const date = new Date(
+// //     new Date(year, month + 1).setDate(new Date(year, month + 1).getDate() - 1)
+// //   );
+// //   return date.getDate();
+//
+//   return new Date(year, month + 1, 0).getDate(); // lol, it works
+// };
+
+// console.log(getLastDayOfMonth(2012, 1));
+
+// Task 8
+
+// const getSecondsToday = () => {
+//   return Math.round(
+//     (new Date().getTime() -
+//       new Date(new Date().setHours(0, 0, 0, 000)).getTime()) /
+//       1000
+//   );
+// };
+
+// console.log(getSecondsToday());
+
+// Task 9
+
+// const getSecondsToTomorrow = () => {
+//   return Math.round(
+//     (new Date(new Date().setHours(24, 0, 0, 000)).getTime() -
+//       new Date().getTime()) /
+//       1000
+//   );
+// };
+
+// console.log(getSecondsToTomorrow());
+
+// Task 10
+
+// const formatDate = date => {
+//   const past = new Date() - new Date(date);
+
+//   const options = {
+//     year: "2-digit",
+//     month: "2-digit",
+//     day: "2-digit",
+//     hour: "numeric",
+//     minute: "numeric"
+//   };
+
+//   if (past / 1000 < 1) {
+//     return "right now";
+//   }
+//   if (past / 1000 / 60 < 1) {
+//     return `${Math.round(past / 1000)} seconds ago`;
+//   }
+//   if (past / 1000 / 60 / 60 < 1) {
+//     return `${Math.round(past / 1000 / 60)} min ago`;
+//   }
```

Comment:
first of all, make a space between logical blocks
```
  if (past / 1000 / 60 < 1) {
    return `${Math.round(past / 1000)} seconds ago`;
  }

   if (past / 1000 / 60 / 60 < 1) {
     return `${Math.round(past / 1000 / 60)} min ago`;
  }
```

Reply:
Secondly, you can reassign the value of the 'past' variable.  It allows you to get rid of unnecessary calculations.

```
 if (past / 1000 < 1) {
    return 'right now';
  }

  past /= 1000;

  if (past / 60 < 1) {
    return `${Math.round(past)} seconds ago`;
  }

  past /= 60;

  if (past / 60 < 1) {
    return `${Math.round(past)} min ago`;
  }
```